Makefile

```diff
@@ -26,77 +25,171 @@ export PRINT_HELP_PYSCRIPT
 
 BROWSER := python -c "$$BROWSER_PYSCRIPT"
 
+.PHONY: help
 help:
 	@python -c "$$PRINT_HELP_PYSCRIPT" < $(MAKEFILE_LIST)
 
-clean: clean-build clean-pyc clean-coverage clean-test clean-docs ## remove all build, test, coverage, docs and Python artifacts
 
+# CLEAN TARGETS
+
+.PHONY: clean-build
 clean-build: ## remove build artifacts
 	rm -fr build/
 	rm -fr dist/
 	rm -fr .eggs/
 	find . -name '*.egg-info' -exec rm -fr {} +
 	find . -name '*.egg' -exec rm -f {} +
 
+.PHONY: clean-pyc
 clean-pyc: ## remove Python file artifacts
 	find . -name '*.pyc' -exec rm -f {} +
 	find . -name '*.pyo' -exec rm -f {} +
 	find . -name '*~' -exec rm -f {} +
 	find . -name '__pycache__' -exec rm -fr {} +
 
+.PHONY: clean-docs
+clean-docs: ## remove previously built docs
+	-$(MAKE) -C docs clean 2>/dev/null  # this fails if sphinx is not yet installed
+
+.PHONY: clean-coverage
 clean-coverage: ## remove coverage artifacts
 	rm -f .coverage
 	rm -f .coverage.*
 	rm -fr htmlcov/
 
-clean-test: ## remove test and coverage artifacts
+.PHONY: clean-test
+clean-test: ## remove test artifacts
 	rm -fr .tox/
 	rm -fr .pytest_cache
 
+.PHONY: clean
+clean: clean-build clean-pyc clean-test clean-coverage clean-docs ## remove all build, test, coverage, docs and Python artifacts
+
+
+# INSTALL TARGETS
+
+.PHONY: install
+install: clean-build clean-pyc ## install the package to the active Python's site-packages
+	pip install .
+
+.PHONY: install-test
+install-test: clean-build clean-pyc ## install the package and test dependencies
+	pip install .[test]
+
+.PHONY: install-develop
+install-develop: clean-build clean-pyc ## install the package in editable mode and dependencies for development
+	pip install -e .[dev]
+
+
+# LINT TARGETS
+
+.PHONY: lint
 lint: ## check style with flake8 and isort
-	flake8 btb tests examples
-	isort -c --recursive btb tests examples
+	flake8 btb tests
+	isort -c --recursive btb tests
 
+.PHONY: fix-lint
 fix-lint: ## fix lint issues using autoflake, autopep8, and isort
-	autopep8 --in-place --recursive --aggressive --aggressive btb tests examples
-	isort --apply --atomic --recursive btb tests examples
+	find btb -name '*.py' | xargs autoflake --in-place --remove-all-unused-imports --remove-unused-variables
```

Comment:
I suggest abandoning autoflake.

and there is no need to re-separate the autopep8 and isort commands, they can be a single command that applies to `btb tests examples`.

Reply:
The need of this change is simply that this is what we have in the cookiecutter-pypackage.
We can always modify it in the future when we change the cookiectter template and port the changes here.